### ISSUE TITLE
e5620

### ISSUE THREAD

**Issue by Terminalstrip** [Piker-Alpha/ssdtPRGen.sh#154]
Hello,
i'm not able to generate SSDT for my system.
I have dual CPU Xeon e5620, i get an error when i try to generate, how to solve?

**Comment by Piker-Alpha** [comment]
I don't have a crystal ball and thus I have no idea what you are doing, but there are a couple of examples here (see other issues) from people who used ssdtPRGen.sh to create SSDT.dsl with a custom definition in Data/User Defined.cfg I would say start there.

**Comment by Terminalstrip** [comment]
Thanks for the reply . I asked you to guess anything but tell me if your 
program is compatible with Xeon processors with socket 1366. It did not 
seem a difficult question.

Il 23/08/15 22:20, Pike ha scritto:

**Comment by Piker-Alpha** [comment]
I don't know what you did. I don't know how you launched the script. I don't know what the error was that you got. Hence the reference to a crystal ball, because I still don't know anything yet I told you to look around in the other issues, because others did it (on their own).

**Comment by Terminalstrip** [comment]
I'm using the latest version of your script .
I launched with these arguments -Turbo 3200

I've try with ./ssdtPRGen.sh -p e5620 -bclk 133 -f 2400 -turbo 3600

I receive everyday this errror message:

ssdtPRGen.sh v0.9 Copyright (c) 2011-2012 by † RevoGirl
             v6.6 Copyright (c) 2013 by † Jeroen
             v15.6 Copyright (c) 2013-2015 by Pike R. Alpha
-----------------------------------------------------------
Bugs > https://github.com/Piker-Alpha/ssdtPRGen.sh/issues <

gDataPath: /Users/TerminalStrip/Library/ssdtPRGen/Data
Override value: (-bclk) frequency, now using: 133 MHz!
Override value: (-f) clock frequency, now using: 2800 MHz!
Override value: (-turbo) maximum (turbo) frequency, now using: 3330 MHz!

System information: Mac OS X 10.9.5 (13F1112)
Brandstring 'Intel(R) Xeon(R) CPU E5620 @ 2.40GHz'

/Users/TerminalStrip/Library/ssdtPRGen/Data/Haswell.cfg: line 85: unexpected EOF while looking for matching `''
/Users/TerminalStrip/Library/ssdtPRGen/Data/Haswell.cfg: line 253: syntax error:
Error: Unknown processor model ...
       Visit http://ark.intel.com to gather the required data:
       Processor Number
       TDP
       Low Frequency Mode (use AppleIntelInfo.kext)
       Base Frequency
       Max Turbo Frequency
       Cores
       Threads
Aborting ...
Done.


MacPro:ssdtPRGen.sh-master TerminalStrip$ 


On the folder DATA i dont find any file User Defined.cfg and if I create it I have the same error.
How to procede?

**Comment by Piker-Alpha** [comment]
Now we are getting somewhere.

1.) rm -R /Library/ssdtPRGen/Data
- Removed old data files.

2.) Download the (latest) Beta.

3.) Add your processor data from ark.intel.com to: /Data/User Defined.cfg (create file).

4.) Launch ./ssdtPRGen.sh with -p E5620 (to pick up the data).

Now tell me what you get.

**Comment by Terminalstrip** [comment]
Ok thanx....
I've made a custom file called User Defined.cfg
I've put it in Data folder
This is the string: 
gUserDefinedCPUList=(
E5620,80,1066,2400,2660,4,8,2,133
)

RESULT: Bugs > https://github.com/Piker-Alpha/ssdtPRGen.sh/issues <

gDataPath: /Users/TerminalStrip/Library/ssdtPRGen/Data

Error: Invalid argument detected: with
Aborting ...
Done.


My question now is if my string are correct, plase let me know

E5620 = model name
80 = TDP
1066 = Ram Type
2400 = Processor Base Freq
2660 = Max Turbo Freq
4 = Numbers of cores
8= Numbers of thread???
2= Numbers of processors???
133= Clock???

**Comment by Terminalstrip** [comment]
YOU CAN SEE MY CONFIG FILE AT 

HTTP://WWW.Terminalstrip.net/Desktop/UserDefined.cfg.zip

**Comment by Piker-Alpha** [comment]
Try this:

gUserDefinedCPUList=(
#Processor Number, TDP, LFM, Clock Speed, Turbo Frequency, Cores, Threads, Bridge, Clock
E5520,80,1600,2400,2660,4,8,2,133
)

**Comment by Terminalstrip** [comment]
See to work but it stop on processor model:

ssdtPRGen.sh v0.9 Copyright (c) 2011-2012 by † RevoGirl
             v6.6 Copyright (c) 2013 by † Jeroen
             v15.6 Copyright (c) 2013-2015 by Pike R. Alpha
-----------------------------------------------------------
Bugs > https://github.com/Piker-Alpha/ssdtPRGen.sh/issues <

gDataPath: /Users/TerminalStrip/Library/ssdtPRGen/Data
Override value: (-p) processor model, now using: E5620!

System information: Mac OS X 10.9.5 (13F1112)
Brandstring 'Intel(R) Xeon(R) CPU E5620 @ 2.40GHz'

Error: Unknown processor model ...
       Visit http://ark.intel.com to gather the required data:
       Processor Number
       TDP
       Low Frequency Mode (use AppleIntelInfo.kext)
       Base Frequency
       Max Turbo Frequency
       Cores
       Threads
Aborting ...
Done.


MacPro:~ TerminalStrip$

**Comment by Piker-Alpha** [comment]
Sorry. Typo. E5520 -> E5620

**Comment by Terminalstrip** [comment]
Hey, thanx for the reply!

i wrote: ssdtPRGen.sh -p E5620 and...

Same error, unknow processor model...ect...

My SMbios is MacPro 5,1, You need anothers info to complete the cfg file?

I need to modify also the ssdtPRGen.sh?
Let me know and thanx for You help

**Comment by Piker-Alpha** [comment]
I don't think that you need -p and no you also don't have to edit ssdtPRGen.sh but the data in "User Defined.cfg" must be correct.

**Comment by Terminalstrip** [comment]
Ok , I keep getting error .
I believe that the problem is an incorrect completion of the cfg file .
I do not know how to find the value of the bridge.
To launch the application I always open a terminal window and I dragged the file ssdtPRGen.sh , what am I doing wrong ?

**Comment by Piker-Alpha** [comment]
What I do is:

1.) open a terminal window (from Utilities).
2.) cd to the folder with ssdtPRGen.sh
3.) enter ./ssdtPRGen.sh

Verify if the correct data is in: ~/Library/ssdtPRGen.sh/Data/User Defined.cfg
If not fix that. 

The bridge type is-cpu type (see -h script argument)

**Comment by Terminalstrip** [comment]
Thank you very much.
Tomorrow i will try 

Gian Luigi Battistini
Tecnico Audio
Via della Misericordia, 47
44124 Ferrara
Tel. +39 0532 470666
Mob. +39 346 9679580
gianluigibattistini@gmail.com
gianluigibattistini@pec.it
info@terminalstrip.net

**Comment by Terminalstrip** [comment]
Same error...
my cpu is Xeon 1366, this processor is not Sandy, Not Ivy, not Haswell 
and not Brodwell. In the script argument i see only this 4 categories 
for cpu-type....
After i will record a video and a send on my server,

So I can show you all the steps and the files in the folder of the script

Il 26/08/15 12:26, Pike ha scritto:

**Comment by Terminalstrip** [comment]
ok, se my video please

http://www.terminalstrip.net/Desktop/SSDT-Try.mov.zip

P.S. I dont have DSDT, i need it to complete the SSDT process?

Gian Luigi Battistini
Tecnico Audio
Via della Misericordia, 47
44124 Ferrara
Tel. +39 0532/470666
Mob. +39 346/9679580
gianluigibattistini@gmail.com
gianluigibattistini@pec.it
info@terminalstrip.net

Il 26/08/15 12:26, Pike ha scritto:

**Comment by Terminalstrip** [comment]
can you have news?

**Comment by Piker-Alpha** [comment]
Please zip ~/Library/ssdtPRGen/User Defined.cfg and e-mail me that file (see script for e-mail address).

**Comment by Terminalstrip** [comment]
attacched

Gian Luigi Battistini
Tecnico Audio
Via della Misericordia, 47
44124 Ferrara
Tel. +39 0532/470666
Mob. +39 346/9679580
gianluigibattistini@gmail.com
gianluigibattistini@pec.it
info@terminalstrip.net

Il 06/09/15 16:24, Pike ha scritto:

**Comment by Terminalstrip** [comment]
script for email address? where?
If You prefere i've uploaded the file here  
http://www.terminalstrip.net/Clover/UserDefined.cfg.zip

**Comment by Piker-Alpha** [comment]
Thanks. I'lll have a look at it tonight.
